File: leetcode/169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        nMap = {}
        for v in nums:
            nMap[v] = nMap.get(v, 0) + 1
        
        n = n // 2
        for element, count in nMap.items():
            if count > n:
                return element
        return  -1

File: leetcode/169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> count;
        int n = nums.size() - 1;
        int check = n / 2;
        int result;
        for (int i : nums) {
            count[i]++;
            if (count[i] > check) {
                result = i;
                break;
            }
        }
        return result;
    }
};